oxidation/libparsec_python/src/crypto.rs

```diff
@@ -52,3 +55,60 @@ impl HashDigest {
         }
     }
 }
+
+#[pyclass]
+#[derive(PartialEq, Eq)]
+pub(crate) struct SecretKey(parsec_api_crypto::SecretKey);
+
+#[pymethods]
+impl SecretKey {
+    #[new]
+    fn new(data: &[u8]) -> PyResult<Self> {
+        match parsec_api_crypto::SecretKey::try_from(data) {
+            Ok(h) => Ok(Self(h)),
+            Err(err) => Err(PyValueError::new_err(err)),
+        }
+    }
+
+    #[classmethod]
+    fn generate(_cls: &PyType) -> PyResult<SecretKey> {
+        Ok(SecretKey(parsec_api_crypto::SecretKey::generate()))
+    }
+
+    #[getter]
+    fn secret<'p>(&self, py: Python<'p>) -> PyResult<&'p PyBytes> {
+        Ok(PyBytes::new(py, self.0.as_ref()))
+    }
+
+    pub fn encrypt<'p>(&self, py: Python<'p>, data: &[u8]) -> Result<&'p PyBytes, PyErr> {
+        Ok(PyBytes::new(py, &self.0.encrypt(data)))
+    }
+
+    pub fn decrypt<'p>(&self, py: Python<'p>, ciphered: &[u8]) -> PyResult<&'p PyBytes> {
+        match self.0.decrypt(ciphered) {
+            Ok(v) => Ok(PyBytes::new(py, &v)),
+            Err(err) => Err(CryptoError::new_err(err)),
+        }
+    }
+
+    pub fn hmac<'p>(
+        &self,
+        py: Python<'p>,
+        data: &[u8],
+        digest_size: usize,
+    ) -> PyResult<&'p PyByteArray> {
+        Ok(PyByteArray::new(py, &self.0.hmac(data, digest_size)))
```

Comment:
Why returning a `PyByteArray`? Do we need the result to be mutable?

Reply:
no right, will change this